lib/semantics/expression.cc

```diff
@@ -2279,10 +2273,12 @@ MaybeExpr ExpressionAnalyzer::ExprOrVariable(const PARSED &x) {
 }
 
 MaybeExpr ExpressionAnalyzer::Analyze(const parser::Expr &expr) {
+  auto save{GetContextualMessages().SetLocation(expr.source)};
```

Comment:
You renamed a similar `save` variable to `restorer` in `AnalyzeCall` so I think that would be a better name here too.

Reply:
Will do here & everywhere else in lib/.